### ISSUE TITLE
1.4-beta0 - Missing endpoints "no endpoints available for service \"kubernetes-dashboard\""

### ISSUE THREAD

**Issue by stevezau** [kubernetes/kubernetes#32099]
This seems similar to https://github.com/kubernetes/kubernetes/issues/23535 but i believe it's unrelated.

**Kubernetes version** 
```
Client Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-beta.0", GitCommit:"44368db9916cc345ebef8b6fbde3cdf0dc9d79dc", GitTreeState:"clean", BuildDate:"2016-09-01T21:26:13Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-beta.0", GitCommit:"44368db9916cc345ebef8b6fbde3cdf0dc9d79dc", GitTreeState:"clean", BuildDate:"2016-09-01T21:16:48Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
s
```

**Environment**:
Deployed using AWS Turnkey solution (kube-up) with 3 nodes on r3.xlarge. Running on ```Linux ip-172-20-0-9 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u1 (2016-09-03) x86_64 GNU/Linux``` 

**What happened**:
I've deployed kubernetes 1.4-beta0 as i need docker 12.1 support. The reason is the current version of docker the ships with 1.3.6 has a bug that effects me (https://github.com/docker/docker/issues/21848) 

When deploying via kube-up.sh it was in a never ending loop waiting for the cluster to come online. From the logs docker was unable to start. I fixed this by manually upgrading the master/minions to docker 12.1 following instructions (https://docs.docker.com/engine/installation/linux/debian/)

The cluster came online but i'm unable to access any endpoints (such as the kube dashboard). I recieve the following
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "no endpoints available for service \"kubernetes-dashboard\"",
  "reason": "ServiceUnavailable",
  "code": 503
}
```

More info

kubectl logs kubernetes-dashboard-v1.1.1-coyi5 --namespace kube-system
```
Starting HTTP server on port 9090
Creating API server client for https://10.0.0.1:443
Successful initial request to the apiserver, version: v1.4.0-beta.0
Creating in-cluster Heapster client
```

```
```

kubectl describe service kubernetes-dashboard --namespace=kube-system
```
Name:			kubernetes-dashboard
Namespace:		kube-system
Labels:			k8s-app=kubernetes-dashboard
			kubernetes.io/cluster-service=true
Selector:		k8s-app=kubernetes-dashboard
Type:			ClusterIP
IP:			10.0.186.17
Port:			<unset>	80/TCP
Endpoints:		<none>
Session Affinity:	None
```

kubectl describe endpoints --namespace=kube-system
```
....TRUNCATED....
Name:		kube-dns
Namespace:	kube-system
Labels:		k8s-app=kube-dns
		kubernetes.io/cluster-service=true
		kubernetes.io/name=KubeDNS
Subsets:
No events.

Name:		kube-scheduler
Namespace:	kube-system
Labels:		<none>
Subsets:
No events.

Name:		kubernetes-dashboard
Namespace:	kube-system
Labels:		k8s-app=kubernetes-dashboard
		kubernetes.io/cluster-service=true
Subsets:
No events.
....TRUNCATED....

```

**What you expected to happen**:
I expect the endpoints to be available.

**How to reproduce it** (as minimally and precisely as possible):
1) Download 1.4-beta0 and deploy to AWS using kube-up.sh
2) Upgrade to docker 12.1
3) Restart master/minions

**Comment by stevezau** [comment]
So it seems the  kubernetes-dashboard service has no endpoints defined.

```
Name:			kubernetes-dashboard
Namespace:		kube-system
Labels:			k8s-app=kubernetes-dashboard
			kubernetes.io/cluster-service=true
Selector:		k8s-app=kubernetes-dashboard
Type:			ClusterIP
IP:			10.0.186.17
Port:			<unset>	80/TCP
Endpoints:		<none>
Session Affinity:	None
```

I tried to connect on one of the minions..  but get

```curl: (7) Failed to connect to 10.0.186.17 port 80: Connection refused```

I'm still pretty new to kubernetes so please point me in the right direction to troubleshoot this issue as i'm flying blind.

**Comment by stevezau** [comment]
Forgot to add pod describe

kubectl describe pod kubernetes-dashboard-v1.1.1-coyi5 --namespace=kube-system
```
Name:		kubernetes-dashboard-v1.1.1-coyi5
Namespace:	kube-system
Node:		ip-172-20-0-16.ec2.internal/172.20.0.16
Start Time:	Tue, 06 Sep 2016 13:45:33 +1000
Labels:		k8s-app=kubernetes-dashboard
		kubernetes.io/cluster-service=true
		version=v1.1.1
Status:		Running
IP:		172.17.0.8
Controllers:	ReplicationController/kubernetes-dashboard-v1.1.1
Containers:
  kubernetes-dashboard:
    Container ID:	docker://b484a40a1b608931e9affc9f79eda654b768efb8e749757f0a1eab51279ba423
    Image:		gcr.io/google_containers/kubernetes-dashboard-amd64:v1.1.1
    Image ID:		docker://sha256:f739d2414b140a0ca907c6e024e109a7e1b3c899cb1e9a9397bda77ca0891c52
    Port:		9090/TCP
    Limits:
      cpu:	100m
      memory:	50Mi
    Requests:
      cpu:		100m
      memory:		50Mi
    State:		Running
      Started:		Tue, 06 Sep 2016 14:28:18 +1000
    Last State:		Terminated
      Reason:		Completed
      Exit Code:	0
      Started:		Tue, 06 Sep 2016 13:46:42 +1000
      Finished:		Tue, 06 Sep 2016 14:26:15 +1000
    Ready:		True
    Restart Count:	1
    Liveness:		http-get http://:9090/ delay=30s timeout=30s period=10s #success=1 #failure=3
    Volume Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-5gv0d (ro)
    Environment Variables:	<none>
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	True 
  PodScheduled 	True 
Volumes:
  default-token-5gv0d:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-5gv0d
QoS Class:	Guaranteed
Tolerations:	CriticalAddonsOnly=:Exists
No events.
```

**Comment by aronchick** [comment]
The IP address you're accessing is internal to Kubernetes, not available at the node level.

https://github.com/kubernetes/dashboard

You should access (via a network that can see the master) https://<kubernetes-master>/ui

Would you mind pasting this to StackOverflow? We try to have frequently asked questions answered there.

**Comment by aronchick** [comment]
The IP address you're accessing is internal to Kubernetes, not available at the node level.

https://github.com/kubernetes/dashboard

You should access (via a network that can see the master) https://<kubernetes-master>/ui

Would you mind pasting this to StackOverflow? We try to have frequently asked questions answered there.

**Comment by aronchick** [comment]
The IP address you're accessing is internal to Kubernetes, not available at the node level.

https://github.com/kubernetes/dashboard

You should access (via a network that can see the master) https://<kubernetes-master>/ui

Would you mind pasting this to StackOverflow? We try to have frequently asked questions answered there.

**Comment by stevezau** [comment]
Hi @aronchick,

It seems this is an issue with 1.4-beta0 when installing docker 1.12.1 as it breaks all the services. Getting no endpoints errors for most of the services.

I raise this here as i thought 1.4 was meant to support 1.12.1 and this sounds like it's a bug/issue. I did a clean install with AWS kube-up.sh and simply upgraded docker.

**Comment by stevezau** [comment]
Update: I just manually rebuild the docker binary including the patch from https://github.com/docker/docker/commit/86002decd0936fa5d466ba7dcbb3fd1524fd86dd

This now allows docker stats to work and also means i don't need to upgrade to docker 1.12.1..

It's a band-aid fix till kube supports 1.12.1
